Use read-only SQLite connection

jan-janssen opened the pull request:
When the `mendeleev` package is located on a Network File System (NFS) Sqlalchemy fails with an [OperationalError](https://docs.sqlalchemy.org/en/20/errors.html#operationalerror). 

```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) disk I/O error
```

This is a [known limitation of SQLite](https://www.sqlite.org/faq.html#q5):
> This is because fcntl() file locking is broken on many NFS implementations. You should avoid putting SQLite database files on NFS if multiple processes might try to access the file at the same time.

As the data in the `elements.db` file never changes, the solution is to connect to the SQLite database using [read-only mode](https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#uri-connections). This is implemented in this pull request. 

Originally reported by @pkruzikova

lmmentel commented:
Thanks @jan-janssen & @pkruzikova :raised_hands: 

In general the changes make sense, however I would rather have a flag that defaults to read-only mode for most cases but still allows to grab the `engine` for making edits and additions to the db programatically. I might be the  only person who's doing that atm but wouldn't want to duplicate the function for the "dev" mode. Could you add a flag to `get_engine` and `get_session` to parametrize the access mode?

There seem to be some small linting errors, that should be easy to fix.

jan-janssen commented:
> Could you add a flag to `get_engine` and `get_session` to parametrize the access mode?

@lmmentel  I added the flag to `get_engine` and `get_session`.

> There seem to be some small linting errors, that should be easy to fix.

In addition I fixed the linting errors.

jan-janssen commented:
Perfect, thanks a lot - @lmmentel Can you create a new release?

lmmentel commented:
Yes :100:
